### ISSUE TITLE
cli: create workflow parser library

### ISSUE THREAD

**Issue by ivotron** [systemslab/popper#617]
strip out the `cli/popper/parser.py` module and create a standalone library. As part of this, we can define a schema using [`pykwalify`](https://github.com/Grokzen/pykwalify) that allows us to automatically validate a given workflow file. For example:

```yaml
type: map
mapping:
  steps:
    type: seq
    sequence:
    - type: map
      mapping:
        uses:
          type: str
          required: yes
        id:
          type: str
        args:
          type: seq
          sequence:
          - type: str
        runs:
          type: seq
          sequence:
          - type: str
        secrets:
          type: seq
          sequence:
          - type: str
        env:
          type: map
          matching-rule: "any"
          mapping:
            regex;(.+):
              type: str
type: map
mapping:
  options:
    type: map
    mapping:
      secrets:
        type: seq
        sequence:
        - type: str
      env:
        type: map
        matching-rule: "any"
        mapping:
          regex;(.+):
            type: str
```

**Comment by JayjeetAtGithub** [comment]
@ivotron Can we use the [go parser](https://github.com/actions/workflow-parser) directly from our code to validate the workflow file?
Please have a look at [this](https://github.com/JayjeetAtGithub/go-parser-python-bindings).

**Comment by ivotron** [comment]
thanks a lot for looking into this @JayjeetAtGithub!

the problem with using python-go bindings is that it'd make the popper python package harder to install, since now there would be a compilation step in the installation process. This can somewhat be aliviated by using wheel's but it is more work on our end. Using a pure-python implementation allows us to not deal with all this.

in addition, `pyhcl` helps us not to start from scratch.

**Comment by JayjeetAtGithub** [comment]
Yes, I didn't think about the installation process.

**Comment by ivotron** [comment]
closed by 23802dabf95cf5a048df1439d4d76cb01781082c
